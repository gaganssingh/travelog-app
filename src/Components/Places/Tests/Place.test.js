jest.mock("react-dom");

import React from "react";
import { createPortal } from "react-dom";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";

class Place extends React.Component {
    constructor(props) {
        super(props);
        this.PlaceContainer = document.createElement("div");
    }
    render() {
        return <div>{createPortal(<div>hello</div>, this.PlaceContainer)}</div>;
    }
}

it("Place renders", () => {
    const component = renderer.create(<Place />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it("Check it returns a div", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Place />);
    const result = renderer.getRenderOutput();

    expect(result.type).toBe("div");
    expect(result).toMatchSnapshot();
});
