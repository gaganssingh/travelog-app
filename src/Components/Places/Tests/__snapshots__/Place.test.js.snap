// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check it returns a div 1`] = `<div />`;

exports[`Place renders 1`] = `<div />`;
